src/VerifyingPaymaster.sol

```diff
@@ -372,7 +373,8 @@ contract VerifyingPaymaster is BasePaymaster, Ownable2Step {
     ///
     /// @return uint256 Token amount
     function _calculateTokenCost(uint256 gasCost, uint256 tokenExchangeRate) internal pure returns (uint256) {
-        return (gasCost * tokenExchangeRate) / 1e18;
+        // Use mul div up so min amount is 1 
+        return FixedPointMathLib.mulDivUp(gasCost, tokenExchangeRate, 1e18);
```

Comment:
Very minor detail: technically if either `gasCost` or `tokenExchangeRate` is zero then the result can still be zero because the un-rounded answer is truly zero. I think this is fine because neither argument will be zero in practice, so this is just an observation

Reply:
Yup sounds good!